### PR TITLE
Fixes #122: Better DateTimeWidget.

### LINKED ISSUE #122: Create a better DateTimeWidget.
Right now a datetime field is handled by a very simple text widget expecting the user to enter a JSON valid datetime.

We should provide a better UX by default, through a more advanced, convenient and usable widget.

### PR DESCRIPTION
**Pull request by n1k0**
Refs #122.

r=? @magopian @leplatrem

### CODE REVIEW

File: playground/samples/simple.js [hunk 1]
```diff
@@ -24,6 +24,11 @@ module.exports = {
         type: "string",
         title: "Password",
         minLength: 3
+      },
+      date: {
+        type:"string",
```
Comment (magopian) [inline playground/samples/simple.js]: missing space after the `:`

File: test/StringField_test.js [hunk 2]
```diff
@@ -209,17 +198,19 @@ describe("StringField", () => {
...
-      });
+      Simulate.change(node.querySelector("#root_year"), {target: {value: "2010"}});
+      Simulate.change(node.querySelector("#root_month"), {target: {value: "12"}});
+      Simulate.change(node.querySelector("#root_day"), {target: {value: "1"}});
+      Simulate.change(node.querySelector("#root_hour"), {target: {value: "0"}});
+      Simulate.change(node.querySelector("#root_minute"), {target: {value: "0"}});
+      Simulate.change(node.querySelector("#root_second"), {target: {value: "0"}});
 
-      expect(node.querySelector("[type=datetime-local]").value).eql(newDatetime);
+      expect(comp.state.formData).eql("2010-12-01T00:00:00.000Z");
```
Comment (magopian) [inline test/StringField_test.js]: Would it make sense to have another test (or this one?) using different (non-zero) values for the time?

File: test/utils_test.js [hunk 3]
```diff
@@ -470,4 +471,23 @@ describe("utils", () => {
...
+          "minute": 1,
+          "second": 30,
+        });
+    });
+
+    it("should raise on invalid JSON datetime", () => {
+      expect(() => parseDateString("plop"))
+        .to.Throw(Error, "Unable to parse");
+    });
+  });
```
Comment (magopian) [inline test/utils_test.js]: No test for `toDateString`?

File: test/StringField_test.js (hunk 2 above)
Comment (magopian) [inline test/StringField_test.js]: Maybe also add a test to make sure that all the options are present in the selects? It would be unusable if for some reason we render selects which are empty, or missing some values...

File: test/StringField_test.js [hunk 4]
```diff
@@ -211,6 +211,15 @@ describe("StringField", () => {
...
       expect(comp.state.formData).eql("2010-12-01T00:00:00.000Z");
+
+      Simulate.change(node.querySelector("#root_year"), {target: {value: "2012"}});
+      Simulate.change(node.querySelector("#root_month"), {target: {value: "10"}});
+      Simulate.change(node.querySelector("#root_day"), {target: {value: "2"}});
+      Simulate.change(node.querySelector("#root_hour"), {target: {value: "1"}});
+      Simulate.change(node.querySelector("#root_minute"), {target: {value: "2"}});
+      Simulate.change(node.querySelector("#root_second"), {target: {value: "3"}});
+
+      expect(comp.state.formData).eql("2012-10-02T01:02:03.000Z");
```
Comment (n1k0) [inline test/StringField_test.js]: I don't get the purpose of this supplementary test, hint?

Also I usually like to have a single assertion per `it` block, much easier to identify the failing use case.

File: test/utils_test.js [hunk 5]
```diff
@@ -490,4 +492,24 @@ describe("utils", () => {
...
+  describe("toDateString()", () => {
+    it("should transform an object to a valid json datetime", () => {
+      expect(toDateString({
+          "year": 2016,
+          "month": 4,
+          "day": 5,
+          "hour": 14,
+          "minute": 1,
+          "second": 30,
+        }))
```
Comment (n1k0) [inline test/utils_test.js]: Indented two spaces too many

File: test/StringField_test.js [hunk 6]
```diff
@@ -240,6 +249,28 @@ describe("StringField", () => {
...
+        12,
+        31,
+        24,
+        60,
+        60
+      ]);
+      const monthOptions = node.querySelectorAll("select#root_month option");
+      const monthOptionsValues = [].map.call(monthOptions, option => option.value);
+      expect(monthOptionsValues).eql([
+        "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"]);
```
Comment (n1k0) [inline test/StringField_test.js]: You could also test for the labels, which should be padded with `0` when < 10.

Comment (magopian) [inline test/StringField_test.js]: Added

File: test/utils_test.js (hunk 3 above)
Comment (magopian) [inline test/utils_test.js]: Added

**Comment by magopian** [comment]
Fixed by https://github.com/mozilla-services/react-jsonschema-form/commit/179ab21ec0241aa759b61f7a1c309ae1a5fece52
